index.go

```diff
@@ -57,6 +57,20 @@ type CopyIndex interface {
 	CopyReader() CopyReader
 }
 
+// EventKind represents an event code that can be used to execute external callbacks
+// for various events in the index.
+type EventKind int
+
+// EventKindIndex is a type of event that is fired when a new document is added to the index.
+var EventKindIndex = EventKind(1)
```

Comment:
This is not good @CascadingRadium - 'cos now we have events declared in two places. Let's avoid this kind of detail bleed across libraries. Let's move this back into `bleve`.

Reply:
Also looking into the bleve change, I don't quite see the need for this at all. So maybe you can simply drop this for now, and the `FireEvent` API below will not take any attributes.